minion/offset_consumer.go

```diff
@@ -121,8 +121,9 @@ func (s *Service) checkIfConsumerLagIsCaughtUp(ctx context.Context) {
 			highWaterMark := partition.Offset - 1
 			consumedOffset := consumedOffsets[partition.Partition]
 			partitionLag := highWaterMark - consumedOffset
-			if partitionLag < 0 {
+			if partitionLag < 0 || consumedOffset == 0 {
```

Comment:
I think that's not going to work because at the very beginning (before we consumed a message at all) `consumedOffset` would be 0.

Reply:
Any ideas how to define what is:
- 0 because we haven't consumed a message at all
- 0 because there's nothing to consume?

Reply:
Maybe I need to fetch low water marks somehow here. This way the check would be `lowWaterMark  == highWaterMark`